share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
```

Comment:
>while commandline -po produces them space-delimited 

It doesn't - it delimits with newline. Presumably whatever you tested this with changed the separator. I'd be interested to know what that was, since we don't want to be leaking \x1e (that "array-separator").